### PR TITLE
DRV-270: Add missing type check functions

### PR DESCRIPTION
**Pull request by trevorsibanda**
- Adds is_*type* family of functions.

### CODE REVIEW

File: tests/test_query.py [hunk 1]
```diff
@@ -1012,6 +1012,117 @@ def test_to_date(self):
...
+    for p in pairs:
+      d = dict()
+      d[p[0]] = query.count(query.filter_(query.lambda_("v", p[1](query.var("v"))), query.var("vals")))
+      q.append(d)
+
+    actual = self._q(query.let({"vals": values}, query.merge({}, q)))
+    self.assertEqual(actual, expected)
+
+  #endregion
+
```
Comment (evbruno) [inline tests/test_query.py:1125]: Why did u take this approach for the tests? It takes more time to understand the test itself than to validate correctness.

Comment (trevorsibanda) [reply tests/test_query.py:1125]: The Go driver tests are similar. Whilst it takes a bit to understand, this test checks all acceptable types on each of the type check functions and does it in one query.

Comment (evbruno) [reply tests/test_query.py:1125]: The tests are missing `credentials` and `bytes` types. Does python support them ?

Comment (trevorsibanda) [reply tests/test_query.py:1128]: added tests for bytes and credentials

**Review by evbruno — APPROVED** [review]

**Pull request merged**